File: AtCoder/ACP/BootCamp4b/Medium100/cpp/C_Be_Together.cpp
#include <algorithm>
#include <iostream>
#include <stdio.h>

#define MAX_N 100
#define INF 1'000'000'000'000'000'000

typedef long long ll;

using namespace std;

int A[MAX_N];

int main() {
  int N;
  cin >> N;
  int up = -100;
  int low = 100;
  for (int i = 0; i < N; ++i) {
    cin >> A[i];
    up = max(up, A[i]);
    low = min(low, A[i]);
  }
  ll res = INF;
  for (int i = low; i <= up; ++i) {
    ll t = 0;
    for (int j = 0; j < N; ++j) {
      t += (i - A[j]) * (i - A[j]);
    }
    res = min(res, t);
  }
  printf("%lld\n", res);
}

File: AtCoder/ACP/BootCamp4b/Medium100/python/C_Be_Together.py
INF = 10 ** 18
N = int(input())
A = list(map(int, input().split()))
up = max(A)
low = min(A)
res = INF
for i in range(low, up + 1):
    t = 0
    for a in A:
        t += (i - a) ** 2
    res = min(res, t)
print(res)
